pkg/build/openbsd.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package build
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"time"
+
+	"github.com/google/syzkaller/pkg/osutil"
+)
+
+type openbsd struct{}
+
+func (ctx openbsd) build(targetArch, vmType, kernelDir, outputDir, compiler, userspaceDir,
+	cmdlineFile, sysctlFile string, config []byte) error {
+	const kernelName = "SYZKALLER"
+	confDir := fmt.Sprintf("%v/sys/arch/%v/conf", kernelDir, targetArch)
+	compileDir := fmt.Sprintf("%v/sys/arch/%v/compile/%v", kernelDir, targetArch, kernelName)
+
+	if err := ctx.configure(confDir, compileDir, kernelName); err != nil {
+		return err
+	}
+
+	if err := ctx.make(compileDir, "all"); err != nil {
+		return err
+	}
+
+	for src, dst := range map[string]string{
+		"obj/bsd":     "kernel",
+		"obj/bsd.gdb": "obj/bsd.gdb",
+	} {
+		src := filepath.Join(compileDir, src)
+		dst := filepath.Join(outputDir, dst)
+		if err := osutil.CopyFile(src, dst); err != nil {
+			return fmt.Errorf("failed to copy %v -> %v: %v", src, dst, err)
+		}
+	}
+
+	return nil
+}
+
+func (ctx openbsd) clean(kernelDir string) error {
+	return ctx.make(kernelDir, "", "clean")
+}
+
+func (ctx openbsd) configure(confDir, compileDir, kernelName string) error {
+	conf := []byte("include \"arch/amd64/conf/GENERIC\"\npseudo-device kcov 1\n")
```

Comment:
But then I have to use literal newlines?

```go
    conf := []byte(`foo
bar
`)
```

Reply:
I think so, are you disgusted by this format? :)
```go
    conf := []byte(`
include "arch/amd64/conf/GENERIC"
pseudo-device kcov 1`)
```

Reply:
That's a strong wrong :) I don't have a strong opinion, if backticks are preferred I'll make sure to update the patch.

Reply:
I personally eschew escaping when I can. Let's see if Dmitry has preference.

Reply:
I don't have strong preference. Ugly both ways. As everything in computer science :)

For multi-line I would do it as follows, to make file contents occupy lines exclusively so that one can freely copy-paste, etc.
```
    conf := []byte(`
include "arch/amd64/conf/GENERIC"
pseudo-device kcov 1
`)
```